.circleci/config.yml

```diff
@@ -74,17 +74,14 @@ jobs:
       - setup_build_environment
       - gradle:
           args: :tag
-      - run:
-          name: Remove trigger tag
-          command: git tag -d release-$(git describe --abbrev=0)
       - gradle:
           args: dockerPushImages
       - add_ssh_keys:
           fingerprints:
             - '51:2e:8c:f6:b6:7b:ac:ac:cc:1b:b8:39:3d:82:e5:38'
       - run:
           name: Update remote tags
-          command: git push origin :refs/tags/release-$(git describe --abbrev=0) refs/tags/$(git describe --abbrev=0)
+          command: git push origin refs/tags/$(git describe --abbrev=0)
```

Comment:
So there are three options I see here.
1. What's basically happening in this PR: apply the version tag, but don't remove the trigger tag until the release process is complete. The problem with this is that in between, the version in a bad state, and any gradle command will break with the two co-existing version tags on the same commit. This would also persist if the helm release job were to fail for whatever reason, and require manual intervention.

2. Leave the trigger tag removal as is, and change the match criteria for the helm publish to match the actual tag rather than the trigger tag

3. Don't mess with the tags until the end of the release process - either applying the new tag or removing the release tag. I think this feels right to me, but the drawback is that in between we either need to get version references from gradle (which understands how to parse the trigger tag), or/and add parsing logic in commands that get the version directly from git like `CHART_VERSION=$(git describe --abbrev=0)`, as the tag there will now be a trigger tag rather than a version tag.

Reply:
I guess a fourth and IMO best option is to make the release job self contained, and run it in a single step. Right now, we have containers with helm, containers with java (for gradle), but not both. I think circleci supports multiple containers in a job, but I haven't tried it out. We also could make our own build image that supports either case.